src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNames.cs

```diff
@@ -0,0 +1,194 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.ConstructorParametersShouldMatchPropertyNamesDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor PropertyRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageProperty,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescription,
+                                                                             isPortedFxCopRule: true,
+                                                                             isDataflowRule: false);
+        internal static DiagnosticDescriptor FieldRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageField,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescription,
+                                                                             isPortedFxCopRule: true,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(PropertyRule, FieldRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(
+                (compilationStartContext) =>
+                {
+                    INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                    if (jsonConstructorAttributeNamedSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                    compilationStartContext.RegisterOperationBlockStartAction(
```

Comment:
I've tried it before using the current approach. The idea was to reduce the overall amount of checks/calls the analyzer should do before matching a correct type (to have as low an impact on the performance as possible). I thought that `RegisterOperationBlockStartAction` in combination with `RegisterOperationAction` + `OperationKind.ParameterReference` may be more performant than checking every `SymbolKind.NamedType`. But I might be wrong because I haven't measured the performance yet.

Reply:
> The idea was to reduce the overall amount of checks/calls the analyzer should do before matching a correct type (to have as low an impact on the performance as possible)

Thanks, that was my concert too, by my understanding operation, `OperationBlockStartAction` will be fired for each method definition, by accounting that methods are included within a type and a type could have many methods i guess there will be more methods than types (i.e `NamedType`). So i assume action registered by  `RegisterSymbolAction(AnalyzeSymbol, SymbolKind.NamedType)` will be fired less than action registered with `RegisterOperationBlockStartAction(...)` therefore less checks will be performed, but i might wrong CC @mavasani 

> `RegisterOperationBlockStartAction` in combination with `RegisterOperationAction` + `OperationKind.ParameterReference` may be more performant than checking every SymbolKind.NamedType

I am not sure if it is counted as a combination of actions, i would assume first `OperationBlockStartAction`  will be fired for each method block where your initial checks will be made then `RegisterOperationAction + OperationKind.ParameterReference` will be fired.

Instead with the `RegisterSymbolAction` you will not need to register any more action, everything you need to check come with the resulting `NamedTypeSymbol`